### PR TITLE
[release-1.9]: backport: fix a race condition

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Lantao Liu <lantaol@google.com>

NOTE: cherry pick.

### CODE REVIEW

**Comment by allencloud** [comment]
We need to setup the travis CI, right? @Starnop

**Comment by Starnop** [comment]
@allencloud Yeah, we need change the branch of cri-tools to setup the travis CI. I will do that until this PR has been merged.

**Comment by fuweid** [comment]
LGTM

**Comment by fuweid** [comment]
@Starnop could we add the travis-CI here before we merge?

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ dist: trusty
 
 go:
   - 1.9.x
+  - 1.10.x
```
Comment (fuweid) [inline .travis.yml]: the change is ok. but we don't need to run it with go1.10. So just keep the change in the scope of fixing the data race.

Comment (Starnop) [reply .travis.yml]: Got it.

**Comment by Starnop** [comment]
Has been done . refer to https://github.com/kubernetes-sigs/cri-tools/pull/396

**Pull request merged**